mlrun/api/crud/model_endpoints.py

```diff
@@ -1034,45 +1182,254 @@ def build_kv_cursor_filter_expression(
+            "label_names": json.dumps(label_names),
+            "endpoint_type": json.dumps(endpoint_type),
+            "children_uids": json.dumps(children_uids),
+            **searchable_labels,
+        }
+        return attributes
 
     @staticmethod
-    def get_access_key(auth_info: mlrun.api.schemas.AuthInfo):
-        access_key = auth_info.data_session
-        if not access_key:
-            raise mlrun.errors.MLRunBadRequestError("Data session is missing")
-        return access_key
+    def _json_loads_if_not_none(field: typing.Any) -> typing.Any:
+        return json.loads(field) if field is not None else None
 
     @staticmethod
-    def _get_batching_interval_param(intervals_list):
-        """Converting each value in the intervals list into a float number. None
-        Values will be converted into 0.0.
+    def get_endpoint_features(
+        feature_names: typing.List[str],
+        feature_stats: dict = None,
+        current_stats: dict = None,
+    ) -> typing.List[mlrun.api.schemas.Features]:
+        """
+        Getting a new list of features that exist in feature_names along with their expected (feature_stats) and
+        actual (current_stats) stats. The expected stats were calculated during the creation of the model endpoint,
+        usually based on the data from the Model Artifact. The actual stats are based on the results from the latest
+        model monitoring batch job.
+
+        param feature_names: List of feature names.
+        param feature_stats: Dictionary of feature stats that were stored during the creation of the model endpoint
+                             object.
+        param current_stats: Dictionary of the latest stats that were stored during the last run of the model monitoring
+                             batch job.
+
+        return: List of feature objects. Each feature has a name, weight, expected values, and actual values. More info
+                can be found under mlrun.api.schemas.Features.
+        """
 
-        example::
-        Applying the function on a scheduling policy that is based on every hour exactly
-        _get_batching_interval_param(intervals_list=[0, '*/1', None])
-        The result will be: (0.0, 1.0, 0.0)
+        # Initialize feature and current stats dictionaries
+        safe_feature_stats = feature_stats or {}
+        safe_current_stats = current_stats or {}
 
+        # Create feature object and add it to a general features list
+        features = []
+        for name in feature_names:
+            if feature_stats is not None and name not in feature_stats:
+                logger.warn("Feature missing from 'feature_stats'", name=name)
+            if current_stats is not None and name not in current_stats:
+                logger.warn("Feature missing from 'current_stats'", name=name)
+            f = mlrun.api.schemas.Features.new(
+                name, safe_feature_stats.get(name), safe_current_stats.get(name)
+            )
```

Comment:
what is the effect of feature missing for those dictionaries?

Reply:
These stats are important for drift detection. In other words, this warning message it just to let the user know that the stats are missing for a specific feature and therefore it won't be considered in the drift analysis.
By the way, both `feaure_stats` and `current_stats` are used for `Feature Analysis` tab in the UI:
![image](https://user-images.githubusercontent.com/101177563/195300282-9028a693-d65a-420c-a7f5-7fd30081b8f1.png)